demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php

```diff
@@ -2114,6 +2118,26 @@ protected function setAuthorizedUsersToProcedure(Procedure $newProcedure, $bluep
         return $newProcedure;
     }
 
+    /**
+     * If orga has the permission, add current orga to authorized planning offices to given procedure.
+     *
+     * @param string $currentUserId
+     *
+     * @return Procedure
+     *
+     * @throws Exception
+     */
+    protected function addCurrentOrgaToPlanningOffices(Procedure $newProcedure, $currentUserId)
+    {
+        $currentUser = $this->userService->getSingleUser($currentUserId);
+
+        if ($this->accessControlPermissionService->hasPermission(AccessControlPermissionService::CREATE_PROCEDURES_PERMISSION, $currentUser->getOrga(), $this->customerService->getCurrentCustomer(), $currentUser->getRoles())) {
+            $newProcedure->addPlanningOffice($currentUser->getOrga());
```

Comment:
You need to ensure that orga has orgaType OPAUTH. Not sure what happens if there are orgas in the planningOffices property that are no planning offices

Reply:
gotcha, would you put that in the AccessPermissionService or in the ProcedureService? I guess in the AcessPermissionService?

Reply:
it is a business logic, but not sure where is the right place to put it

Reply:
hey @mussbach 7b4032cbf94d4ae086424ff8e8271161e3a7456c I tried to make general this: Check permission if it is allowed for the orga - customer. What do you think? does it make sense? otherwise I can make the check only when adding the planing offices

Reply:
> You need to ensure that orga has orgaType OPAUTH. Not sure what happens if there are orgas in the planningOffices property that are no planning offices
+1 for that. Unfortunately it seems that the Type isn't checked in other cases :(
I suggest to solve this kind of typecheck by an actual strict use of a specific type.
Which means a own class for each orgatype (extending a kind of base-orga-class) But this isn't the PR to do this.

Anyway I do not see the check for the orga type, do I miss it?

Reply:
@mussbach @martendemos 
- here the commit for checking exactly only that orga type and permission combination: 
9ca77e869f5f9fcba11dea1d0116d185db0bb68f

In the commit you will see this line:  ` return in_array(OrgaTypeInterface::PLANNING_AGENCY, $orga->getTypes($this->globalConfig->getSubdomain(), true));`
I decided to check the subdomain and also if the orga is approved. as I understand an orga can be different types based on the customer

- here a commit for adding missed factories for the tests: 6bf7755f9e180fe6ca8c9d0330ad83da80b371de